source/Calamari.Shared/Deployment/Conventions/ConfigurationVariablesConvention.cs

```diff
@@ -7,25 +7,38 @@
 namespace Calamari.Deployment.Conventions
 {
     public class ConfigurationVariablesConvention : IInstallConvention
+    {
+        readonly ConfigurationVariablesService service;
+
+        public ConfigurationVariablesConvention(ConfigurationVariablesService service)
+        {
+            this.service = service;
+        }
+
+        public void Install(RunningDeployment deployment)
+        {
+            service.Install(deployment, deployment.Variables.GetStrings(SpecialVariables.AppliedXmlConfigTransforms, '|'));
+        }
+    }
+
+    public class ConfigurationVariablesService
     {
         readonly ICalamariFileSystem fileSystem;
         readonly IConfigurationVariablesReplacer replacer;
 
-        public ConfigurationVariablesConvention(ICalamariFileSystem fileSystem, IConfigurationVariablesReplacer replacer)
+        public ConfigurationVariablesService(ICalamariFileSystem fileSystem, IConfigurationVariablesReplacer replacer)
         {
             this.fileSystem = fileSystem;
             this.replacer = replacer;
         }
 
-        public void Install(RunningDeployment deployment)
+        public void Install(RunningDeployment deployment, List<string> appliedAsTransforms)
         {
             if (deployment.Variables.GetFlag(SpecialVariables.Package.AutomaticallyUpdateAppSettingsAndConnectionStrings) == false)
             {
                 return;
             }
 
-            var appliedAsTransforms = deployment.Variables.GetStrings(SpecialVariables.AppliedXmlConfigTransforms, '|');
```

Comment:
There's a slight different in behaviour in that we now always read `SpecialVariables.AppliedXmlConfigTransforms` regardless of the `SpecialVariables.Package.AutomaticallyUpdateAppSettingsAndConnectionStrings` flag